internal/storage/postgreSQL/sensorSQL/mqtt.go

```diff
@@ -0,0 +1,142 @@
+package sensorSQL
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage/postgreSQL/entities/sensor"
+)
+
+type SensorRepository struct {
+	db *sql.DB
+}
+
+func NewSensorRepository(db *sql.DB) *SensorRepository {
+	return &SensorRepository{db: db}
+}
+
+func (r *SensorRepository) Insert(ctx context.Context, data *sensor.MqttEntity) (*sensor.MqttEntity, error) {
```

Comment:
Please take and return the domain entity. Only the repo should know about the storage entity, so you need to map it here to the domain entity, this is where the mapper comes in. Take a look at the old mongodb implementation

```suggestion
func (r *SensorRepository) Insert(ctx context.Context, data *sensor.MqttPayload) (*sensor.MqttPayload, error) {
```